### ISSUE TITLE
support the loop and showcontrols=false options for movie playback

### ISSUE THREAD

**Issue by barak** [pdfpc/pdfpc#207]
Forwarded from https://bugs.debian.org/853097
by Francesco Poli (wintermute) <invernomuto@paranoici.org>

----

As the documentation itself states (see section *Movies* of the pdfpc(1)
man page), movies embedded in LaTeX with the `multimedia` package are
played back by pdfpc. For instance, the following LaTeX statement
```latex
  \movie[loop,showcontrols=false]{\includegraphics[width=0.5\textwidth]{figs/mymovie_first_frame}}{mymovie.avi}
```
embeds `mymovie.avi` into a rectangular area (within the document page) that
is filled with the `figs/mymovie_first_frame.jpg` image (the first frame
of `mymovie.avi`), whenever the movie is not played.
Of course, the preamble must include
````latex
  \usepackage{multimedia}
````
otherwise the `\movie` command won't be recognized by pdflatex...

The playback of the embedded movie works flawlessly with pdfpc, except
that the `loop` option is ignored (hence the movie is played only once)
and the `showcontrols=false` option, which is the default, is also not
taken into account (hence the progress bar is always shown).
This is confirmed by the pdfpc(1) man page.

For my needs, the support for the `loop` and `showcontrols=false` options
is really important.

The man page suggests an interesting alternative: using the following
statement:
````latex
  \href{run:mymovie.avi?loop&noprogress}{\includegraphics[width=0.5\textwidth]{figs/mymovie_first_frame}}
````
after having (directly or indirectly) loaded the `hyperref` package:
````latex
  \usepackage{hyperref}
````
This works perfectly in pdfpc, but has the non-negligible flaw of being
specific to pdfpc: it fails to work with any other PDF viewer, which
complains about not finding file `mymovie.avi?loop&noprogress` ...
In other words, this approach, while working with pdfpc, defeats the
portability of the resulting PDF file: this is not great.

In summary, I would like to see the support for the `loop` and
`showcontrols=false` options implemented in the movie playback capability
of pdfpc.

**Comment by AndreasBilke** [comment]
I pushed a new commit where `showcontrols=false` is now respected. But, as far as I can see, the `loop` flag is not exposed via the poppler-glib library. Therefore the `loop` request is a won't fix, since we cannot do a thing about it.

(The QT bindings have a API for accessing the loop feature, the glib bindings don't have this.)

**Comment by barak** [comment]
Should we file a feature request against poppler-glib?

**Comment by AndreasBilke** [comment]
Could do file it? We already had a situation where the glib bindings lacks
features compared to qt (see #125).

**Comment by barak** [comment]
I can if you want. But I have not written to the API so I think someone who actually has, and can make concrete API suggestions and evaluate alternative ideas for providing the functionality would be better. Also should this be combined with #125 into a single feature request? I dunno!

**Comment by jaliste** [comment]
Just FYI, currently poppler lacks developers on the glib side, since the only "big" consumer of this api is evince (and its several forks). So please try to work on a patch for poppler-glib. If the qt5 bindings already have the feature, it shouldn't be so difficult to copy the code and adapt it to be in a gtk/gobject flavor. If you just file a bug in poppler glib, it will stagnate for a long time, while if you provide a patch and then ping Carlos (poppler-glib's mantainer) you will have much better chances of get this feature soon (poppler has a very short release period now). If you have questions or want to discuss your patch before submitting it to review  feel free to contact me.

**Comment by AndreasBilke** [comment]
I reopen this since poppler 0.54 has now the "loop" flag exposed via movie annotations. We should wait until vala updates their bindings.
